### PR TITLE
For #19055: Fix Button Arrangement Credit Card Editor Fragment

### PR DESCRIPTION
**Pull request by kglazko**
### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Comment by kglazko** [comment]
Accidentally created as a real PR instead of a draft PR. Changed to draft.

**Comment by kglazko** [comment]
Here it is when you add padding to the linearlayout
ScrollLayoutWithPadding

**Comment by kglazko** [comment]
Spoke to @jonalmeida, he said "Yeah, that should be fine. Just fix the indentation. In other cases, try using ConstraintLayout as the base layout. There is less need for nested layout files then. :ok_hand: That might take away the need for the ScrollView, I think. But it's a larger change than what this bug is trying to fix." Will investifate ConstraintLayout.

File: app/src/main/res/layout/fragment_credit_card_editor.xml [hunk 1]
```diff
@@ -1,199 +1,204 @@
-<?xml version="1.0" encoding="utf-8"?>
-<!-- This Source Code Form is subject to the terms of the Mozilla Public
+<?xml version="1.0" encoding="utf-8"?><!-- This Source Code Form is subject to the terms of the Mozilla Public
```
Comment (gabrielluong) [inline app/src/main/res/layout/fragment_credit_card_editor.xml:1]:
```suggestion
<?xml version="1.0" encoding="utf-8"?>
<!-- This Source Code Form is subject to the terms of the Mozilla Public
```

**Comment by gabrielluong** [comment]
Lint Error 

[task 2021-04-21T20:49:24.480Z]   Errors found:
[task 2021-04-21T20:49:24.480Z]   
[task 2021-04-21T20:49:24.480Z]   /builds/worker/checkouts/src/app/src/main/res/layout/fragment_credit_card_editor.xml:9: Error: Unused namespace declaration xmlns:android; already declared on the root element [UnusedNamespace]
[task 2021-04-21T20:49:24.480Z]       <LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"

File: app/src/main/res/layout/fragment_credit_card_editor.xml [hunk 2]
```diff
@@ -2,198 +2,204 @@
...
+        xmlns:app="http://schemas.android.com/apk/res-auto"
+        xmlns:tools="http://schemas.android.com/tools"
         android:layout_width="match_parent"
         android:layout_height="wrap_content"
-        android:textColor="?primaryText"
-        app:errorEnabled="true"
-        app:hintEnabled="false">
+        android:layout_margin="16dp"
+        android:orientation="vertical"
+        android:paddingBottom="80dp">
```
Comment (gabrielluong) [inline app/src/main/res/layout/fragment_credit_card_editor.xml:16]: Why did we add this paddingBottom?

Comment (kglazko) [reply app/src/main/res/layout/fragment_credit_card_editor.xml:16]: Because otherwise the buttons would hit the very bottom. It doesn't have to be 80 but it didn't seem to affect anything either way. Just need some padding between the buttons and the edge of the screen.

File: app/src/main/res/layout/fragment_credit_card_editor.xml [hunk 3]
```diff
@@ -2,198 +2,203 @@
...
+            android:paddingStart="3dp"
+            android:paddingEnd="0dp"
+            android:text="@string/credit_cards_expiration_date"
+            android:textColor="?primaryText"
+            android:textSize="12sp"
+            app:fontFamily="@font/metropolis_semibold" />
 
         <LinearLayout
-            android:layout_width="0dp"
+            android:layout_width="match_parent"
```
Comment (gabrielluong) [inline app/src/main/res/layout/fragment_credit_card_editor.xml:111]: Is this change needed?

File: app/src/main/res/layout/fragment_credit_card_editor.xml [hunk 4]
```diff
@@ -2,198 +2,203 @@
...
-        android:letterSpacing="0.05"
-        app:fontFamily="@font/metropolis_semibold" />
-
-    <com.google.android.material.textfield.TextInputLayout
-        android:id="@+id/card_number_layout"
+    android:layout_height="match_parent">
+
+    <LinearLayout
+        xmlns:app="http://schemas.android.com/apk/res-auto"
+        xmlns:tools="http://schemas.android.com/tools"
```
Comment (gabrielluong) [inline app/src/main/res/layout/fragment_credit_card_editor.xml:11]: This should probably be moved into the ScrollView

File: app/src/main/res/layout/fragment_credit_card_editor.xml [hunk 5]
```diff
@@ -2,198 +2,203 @@
...
         android:layout_width="match_parent"
         android:layout_height="wrap_content"
-        android:textColor="?primaryText"
-        app:errorEnabled="true"
-        app:hintEnabled="false">
+        android:layout_margin="16dp"
```
Comment (gabrielluong) [inline app/src/main/res/layout/fragment_credit_card_editor.xml:14]: We should see if this should be in the ScrollView or LinearLayout now that the LinearLayout is no longer the parent container.

**Review by gabrielluong — COMMENTED** [review]
Running into the issue where it doesn't bottom out. See screenshot. I think the solution is to move layout margin from LinearLayout.

Screen Shot 2021-04-27 at 11 28 52 AM

**Comment by gabrielluong** [comment]
@kglazko Just in case you didn't see this. I think this needs 2 changes to fix the issues:
- Moving the `xmlns:app` and `xmlns:tools` into the `ScrollView` since that is the parent container in the file.
- Moving the `android:layout_margin="16dp"` into the `ScrollView` will also resolve the issue I pointed out in the screenshot.

**Comment by kglazko** [comment]
Thank you, @gabrielluong . I will get those over to you today.

**Comment by kglazko** [comment]
Ok, I implemented your requested changes and am tagging again for review.

**Comment by gabrielluong** [comment]
Will look at this tomorrow.

**Review by gabrielluong — APPROVED** [review]
